### PR TITLE
Prevent too large a number of shares

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#7488

*Description of changes:*
- added shares max digits rule (as per db schema)
- updated unit tests
- updated app version

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/CreateShareStructure/ShareStructure.vue [hunk 1]
```diff
@@ -217,9 +217,10 @@ export default class ShareStructure extends Mixins(CurrencyLookupMixin) {
     let rules: Array<Function> = []
     if (!this.hasNoMaximumShares) {
       rules = [
-        (v: string) => (v !== null && v !== '') || 'Number of shares is required',
+        (v: string) => !!v || 'Number of shares is required',
```
Comment (severinbeauvais) [inline src/components/CreateShareStructure/ShareStructure.vue:220]: It turns out `!!v` works fine if v is a string (ie, "0" is not falsey).

File: src/components/CreateShareStructure/ShareStructure.vue [hunk 2]
```diff
@@ -217,9 +217,10 @@ export default class ShareStructure extends Mixins(CurrencyLookupMixin) {
...
         (v: string) => /^-?\d+$/.test(v) || 'Must be a whole number',
-        (v: string) => +v > 0 || 'Number must be greater than 0'
+        (v: string) => (+v > 0) || 'Number must be greater than 0',
+        (v: string) => (v && v.toString().length < 16) || 'Number must be less than 16 digits'
```
Comment (severinbeauvais) [inline src/components/CreateShareStructure/ShareStructure.vue:223]: I had to cast v to a string as we sometimes set the value to a number in the unit tests.

**Review by cameron-freshworks — APPROVED** [review]
Looks good sir! 
Is this something already implemented in the Shared Components? 
Any concerns of this component being out of sync with the other?

**Review by andrepestana-aot — APPROVED** [review]
LGTM!

**Comment by severinbeauvais** [comment]
> Looks good sir!
> Is this something already implemented in the Shared Components?
> Any concerns of this component being out of sync with the other?

I will implement this in the shared component next. (Yesterday I did it the other way around, but this way is easier as I have an app I can play in.)

This component is mutually exclusive with the one in the shared library. Actually they do the same thing but are not a drop-in replacement for each other... so duplication today, /bcgov/entity#7397 tomorrow!

**Comment by cameron-freshworks** [comment]
> > Looks good sir!
> > Is this something already implemented in the Shared Components?
> > Any concerns of this component being out of sync with the other?
> 
> I will implement this in the shared component next. (Yesterday I did it the other way around, but this way is easier as I have an app I can play in.)
> 
> This component is mutually exclusive with the one in the shared library. Actually they do the same thing but are not a drop-in replacement for each other... so duplication today, /[bcgov/entity#7397](https://github.com/bcgov/entity/issues/7397) tomorrow!

Sounds good 😁

**Pull request merged**